src/nodes/leaf.rs

```diff
@@ -14,6 +14,7 @@ use alloc::vec::Vec;
 /// data associated with the full key. When searching the trie for a specific key, reaching a leaf
 /// node means that the search has successfully found the value associated with that key.
 #[derive(PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
this seems okay, although this will serialize the value as array, should be okay